### PR TITLE
Finish up the per-region sales report admin tool

### PR DESCRIPTION
**Pull request by gauthierm**
It can easily be subclassed to hide/show shipping and to hide/show details for specific countries.

### CODE REVIEW

File: Store/admin/components/SalesByRegionReport/Details.php [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+				sprintf(
+					'Unable to load report with id of “%s”',
+					$id
+				)
+			);
+		}
+
+		$this->start_date = new SwatDate('now', $this->app->default_time_zone);
+		$this->start_date->toUTC();
+		if ($this->start_date->setDate($id, 1, 1) === false) {
```
Comment (isagrant) [inline Store/admin/components/SalesByRegionReport/Details.php:74]: Needs to clear the time as well.

File: Store/admin/components/SalesByRegionReport/Details.php [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+			$this->show_shipping;
+	}
+
+	// }}}
+	// {{{ protected function getTableModel()
+
+	protected function getTableModel(SwatView $view)
+	{
+		$end_date = clone $this->start_date;
+		$end_date->addYears(1);
```
Comment (isagrant) [inline Store/admin/components/SalesByRegionReport/Details.php]: `addYears(1)` will work for a leap year right?

File: Store/admin/components/SalesByRegionReport/Details.php [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+		return $store;
+	}
+
+	// }}}
+	// {{{ protected function getProvStateModel()
+
+	protected function getProvStateModel($country_id)
+	{
+		$end_date = clone $this->start_date;
+		$end_date->addYears(1);
```
Comment (isagrant) [inline Store/admin/components/SalesByRegionReport/Details.php]: `addYears(1)` will work for a leap year right?

File: Store/admin/components/SalesByRegionReport/include/StoreSalesByRegionGroup.php [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+		if ($visible_renderer_count === 0) {
+			return;
+		}
+
+		$tr_tag = new SwatHtmlTag('tr');
+		$tr_tag->class = 'swat-table-view-group';
+		$tr_tag->open();
+
+		$first = true;
+		$second = true;
```
Comment (isagrant) [inline Store/admin/components/SalesByRegionReport/include/StoreSalesByRegionGroup.php:48]: Add comments to explain this madness.

File: Store/admin/components/SalesByRegionReport/include/StoreSalesByRegionGroup.php [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+				$td_tag->colspan = $this->view->getXhtmlColspan() -
+					$visible_renderer_count + 1;
+			}
+
+			$td_tag->open();
+			$renderer->render();
+			$td_tag->close();
+
+			if ($second) {
+				$second = false;
```
Comment (isagrant) [inline Store/admin/components/SalesByRegionReport/include/StoreSalesByRegionGroup.php]: Might as well simplify by sticking this in the early `if ($second) {` block.

**Comment by isagrant** [comment]
`Shipping` column should come before `Gross Sales` column.

**Comment by gauthierm** [comment]
> Shipping column should come before Gross Sales column.

Done

**Comment by isagrant** [comment]
Should show $0 in sales for any `Provstate` that don't have sales in that year (but not countries).

**Pull request merged**